idrstream/download.py

```diff
@@ -0,0 +1,128 @@
+
+    Attributes
+    ----------
+    aspera_path : pathlib.Path
+        path to aspera sdk
+    aspera_key_path : pathlib.Path
+        path to aspera openssh key
+    screens : pd.Dataframe
+        dataframe with curated IDR screen data for specific study
+    idr_id: str
+        accession ID for IDR study
+
+    Methods
+    -------
+    get_IDR_image_path(plate, well_num)
+        get image path for Aspera download by finding screen for specific image in screen metadata
+    download_image(plate, well_num)
+        download image corresponding to plate and well number with Aspera
+
+    Example
+    -------
+    import pathlib
+    from IDR_stream.download import AsperaDownloader
+
+    aspera_path = pathlib.Path("/home/user/.aspera/ascli/sdk/ascp")
+    aspera_key_path = pathlib.Path("asperaweb_id_dsa.openssh")
+    screens_path = pathlib.Path("idr0013-screenA-plates.tsv")
+    idr_id = "idr0013"
+    save_dir = pathlib.Path("downloaded_images/")
+
+    downloader = AsperaDownloader(aspera_path, aspera_key_path, screens_path, idr_id)
+    downloader.download_image("LT0001_02", 4, save_dir)
+    """
+
+    def __init__(
+        self,
+        aspera_path: pathlib.Path,
+        aspera_key_path: pathlib.Path,
+        screens_path: pathlib.Path,
+        idr_id: str,
+    ):
+        """
+        __init__ function for AsperaDownloader class
+
+        Parameters
+        ----------
+        aspera_path : pathlib.Path
+            path to aspera sdk, ex pathlib.Path("/home/user/.aspera/ascli/sdk/ascp")
+        aspera_key_path : pathlib.Path
+            path to aspera openssh key
+        screens_path : pathlib.Path
+            path to tsv with curated IDR screen data for specific study
+        idr_id : str
+            accession ID for IDR study, ex "idr0013"
+        """
+        self.aspera_path = aspera_path
+        self.aspera_key_path = aspera_key_path
+        self.screens = pd.read_csv(
+            screens_path, sep="\t", header=None, names=["Plate", "Screen"]
+        )
+        self.idr_id = idr_id
+
+    def get_IDR_image_path(self, plate: str, well_num: int) -> str:
+        """
+        get image path for Aspera download by finding screen for specific image in screen metadata
+
+        Parameters
+        ----------
+        plate : str
+            name of plate with desired image data
+        well_num : int
+            number of well with desired image data
+
+        Returns
+        -------
+        str
+            path to IDR location of image data for use by Aspera
+        """
+        # get location of screen
+        screen_loc = (
+            self.screens.loc[self.screens["Plate"] == plate, "Screen"]
+            .item()
+            .replace("../screens/", "")
+            .replace(".screen", "")
+        )
+        well = str(well_num).zfill(3)
+        image_path = (
+            f"20150916-mitocheck-analysis/mitocheck/{screen_loc}/hdf5/00{well}_01.ch5"
```

Comment:
to prepare for modularity, I think it's best if you abstract all hardcoded variables to mitocheck-specific data into function arguments.

This will help future IDRstream developers.

Reply:
Because every line in the `get_IDR_image_path` function (that you comment on above) is specific to the mitocheck data I opted to change the name of the function to `get_IDR_mitocheck_movie_path`. I think this will make it very clear to future developers what code needs to be changed for adaptation to other studies. I also did this with other mitocheck-specific functions (like `load_mitocheck_movie_data`).